node/src/node_configurator/configurator.rs

```diff
@@ -1695,6 +1672,114 @@ mod tests {
+                mnemonic_phrase_language: "English".to_string(),
+                mnemonic_passphrase_opt: None,
+            }),
+            consuming_derivation_path_opt: None,
+            consuming_private_key_opt: Some(consuming_private_key.clone()),
+            earning_derivation_path_opt: None,
+            earning_address_opt: Some(earning_address.clone()),
+        };
+        let set_wallet_info_params_arc = Arc::new(Mutex::new(vec![]));
+        let mut persistent_config: Box<dyn PersistentConfiguration> = Box::new(
+            make_default_persistent_configuration()
+                .check_password_result(Ok(true))
+                .set_wallet_info_params(&set_wallet_info_params_arc)
+                .set_wallet_info_result(Ok(())),
+        );
+
+        let result =
+            Configurator::unfriendly_handle_recover_wallets(msg, 1234, &mut persistent_config);
+
+        assert_eq!(result.is_ok(), true);
+        let set_wallet_info_params = set_wallet_info_params_arc.lock().unwrap();
+        assert_eq!(
+            *set_wallet_info_params,
+            vec![(consuming_private_key, earning_address, db_password)]
+        )
+    }
+
+    #[test]
+    fn unfriendly_handle_recover_wallets_handles_no_seed_spec_and_only_earning_wallet_address() {
+        let msg = UiRecoverWalletsRequest {
+            db_password: "password".to_string(),
+            seed_spec_opt: None,
+            consuming_derivation_path_opt: None,
+            consuming_private_key_opt: None,
+            earning_derivation_path_opt: None,
+            earning_address_opt: Some("0x0123456789012345678901234567890123456789".to_string()),
+        };
+        let mut persistent_config: Box<dyn PersistentConfiguration> =
+            Box::new(make_default_persistent_configuration().check_password_result(Ok(true)));
+
+        let result =
+            Configurator::unfriendly_handle_recover_wallets(msg, 1234, &mut persistent_config);
+
+        assert_eq! (result, Err((MISSING_DATA, "If you supply no seed information, you must supply both consuming wallet private key and earning wallet address".to_string())));
+    }
+
+    #[test]
+    fn unfriendly_handle_recover_wallets_handles_seed_but_nothing_about_consuming_wallet() {
+        let db_password = "password".to_string();
+        let earning_address = "0x0123456789012345678901234567890123456789".to_string();
+        let msg = UiRecoverWalletsRequest {
+            db_password: db_password.clone(),
+            seed_spec_opt: Some(UiRecoverSeedSpec {
+                mnemonic_phrase: make_meaningless_phrase_words(),
+                mnemonic_phrase_language: "English".to_string(),
+                mnemonic_passphrase_opt: None,
+            }),
+            consuming_derivation_path_opt: None,
+            consuming_private_key_opt: None,
+            earning_derivation_path_opt: None,
+            earning_address_opt: Some(earning_address.clone()),
```

Comment:
I've noticed that you're sometimes cloning the strings representing the wallet parameters, all despite you may not use the string again if I see well. I can see it in more than one tests around here. But I don't know why Clippy wouldn't tell you...maybe it cannot do that though.